gcc7: Removed java as a target language

Synthetica9 opened the pull request:
###### Motivation for this change

Retry of the... less than overwhelming success that  #39917 was. 

For your convenience: 

> As discussed in  #39866
> 
> GCC6 is the last GCC release to support Java as a target language. I quote @gebner (https://github.com/NixOS/nixpkgs/pull/39866#discussion_r185733458):
> 
> > I think we should remove them completely. AFAICT the only place where we override enableJava et al. is in the gcj6 attribute (which uses gcc6).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Synthetica9 committed:
gcc7: Removed java as a target language

Synthetica9 commented:
CC @Profpatsch @xeji

xeji commented:
Let's try once if we get ofborg to build this (may time out)

@GrahamcOfBorg  build gcc7

Synthetica9 commented:
Probably _will_ time out, took more than an hour on my machine. (Doesn't ofborg give timeout messages btw?)

xeji commented:
I think it does time out after 1hr.

Synthetica9 committed:
gcc7: Removed X11 components, since they were only used by langJava

See also:

https://github.com/NixOS/nixpkgs/pull/39866/commits/51ac1d8df08d606ae5d8af95073bb74e330249f3
https://github.com/NixOS/nixpkgs/pull/39866#discussion_r185813776

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39924&attempt_id=71e7583a-11ad-4298-ad95-889d1665fac7)

Attempted: gcc7

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/bigafakb6jm4mnj2n8rbsghwxifmpxqx-perl-5.24.3.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/dfxsyz4c918c1yaz8wkq7lx70r6k3lnc-which-2.21.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/gwh4jlrk38gwn0z7il0sahvbss017kfm-gmp-6.1.2.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/f8ysrhdpgm4c5pia4sklwmwm1j40z0fy-libelf-0.8.13.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/8i9b54pa1823lsf9rms8p7isna95z59r-isl-0.17.1.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/05nn4j905zvl1cdmfgsszmkza1i3f6zw-mpfr-4.0.1.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/685wawb90iz32nbfnv3624dkbgymj7i1-libmpc-1.1.0.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/1fpi6hbn3w81lx71zi4bbchljniy966d-gcc-7.3.0.drv': 13 dependencies couldn't be built
cannot build derivation '/nix/store/h7bk0h9xgklmbgjdljnlw140av4hsf6l-gcc-wrapper-7.3.0.drv': 7 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/h7bk0h9xgklmbgjdljnlw140av4hsf6l-gcc-wrapper-7.3.0.drv' failed
```
</p></details>

Synthetica9 commented:
Is that just the timeout message?

xeji commented:
No, some dependencies failed, seems unrelated to this PR.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39924&attempt_id=6c67bab4-fc59-4ed0-9242-6adaee5357df)

Attempted: gcc7

<details><summary>Partial log (click to expand)</summary><p>

```
no configure script, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/3a0sxq113fr4jk996h2l10f4612m47p5-gcc-wrapper-7.3.0
strip is /nix/store/pjfm07gclyzz5csx7kqdr4ji2gjzwizk-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/3a0sxq113fr4jk996h2l10f4612m47p5-gcc-wrapper-7.3.0/bin
patching script interpreter paths in /nix/store/3a0sxq113fr4jk996h2l10f4612m47p5-gcc-wrapper-7.3.0
checking for references to /build in /nix/store/3a0sxq113fr4jk996h2l10f4612m47p5-gcc-wrapper-7.3.0...
strip is /nix/store/pjfm07gclyzz5csx7kqdr4ji2gjzwizk-binutils-2.30/bin/strip
strip is /nix/store/pjfm07gclyzz5csx7kqdr4ji2gjzwizk-binutils-2.30/bin/strip
```
</p></details>

Ericson2314 commented:
Don't forget gcc-snapshot

Synthetica9 committed:
gcc-snapshot: Removed java as a target language

Synthetica9 commented:
@Ericson2314 :heavy_check_mark:
